src/overlay/Peer.cpp

```diff
@@ -93,6 +93,34 @@ CapacityTrackedMessage::CapacityTrackedMessage(std::weak_ptr<Peer> peer,
     {
         mMaybeHash = xdrBlake2(msg);
     }
+
+    if (mMsg.type() == TRANSACTION)
+    {
+        auto transaction = TransactionFrameBase::makeTransactionFromWire(
+            self->mAppConnector.getNetworkID(), mMsg.transaction());
+        StellarMessage txMsg;
+        txMsg.type(TRANSACTION);
+        txMsg.transaction() = mMsg.transaction();
+        transaction->getFullHash();
+        transaction->getContentsHash();
+        mTxsMap[*mMaybeHash] = transaction;
+    }
+#ifdef BUILD_TESTS
+    else if (mMsg.type() == TX_SET && OverlayManager::isFloodMessage(mMsg))
+    {
+        StellarMessage txMsg;
+        txMsg.type(TRANSACTION);
+        for (auto const& tx : mMsg.txSet().txs)
+        {
+            auto transaction = TransactionFrameBase::makeTransactionFromWire(
+                self->mAppConnector.getNetworkID(), tx);
+            txMsg.transaction() = tx;
+            transaction->getFullHash();
+            transaction->getContentsHash();
+            mTxsMap[xdrBlake2(txMsg)] = transaction;
```

Comment:
Nitpick: It might be easier to read this if it's deduplicated with the `TRANSACTION` case. Perhaps a lot of this can be refactored out into a common function?

Reply:
Good idea, refactored this.